homeassistant/components/version/binary_sensor.py

```diff
@@ -0,0 +1,61 @@
+"""Binary sensor platform for Version."""
+from __future__ import annotations
+
+from awesomeversion import AwesomeVersion
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, __version__ as HA_VERSION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_SOURCE, DEFAULT_NAME, DOMAIN
+from .coordinator import VersionDataUpdateCoordinator
+from .entity import VersionEntity
+
+HA_VERSION_OBJECT = AwesomeVersion(HA_VERSION)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up version binary_sensors."""
+    coordinator: VersionDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    if (source := config_entry.data.get(CONF_SOURCE)) == "local":
```

Comment:
Is source optional in the config entry data? If not:
```suggestion
    if (source := config_entry.data[CONF_SOURCE]) == "local":
```

Reply:
It should not be, but have to check.